File: src/main/java/com/koldyr/csv/db/PlainSQLGenerator.java
package com.koldyr.csv.db;

import org.apache.commons.lang.StringUtils;

/**
 * Description of class PlainSQLGenerator
 *
 * @created: 2018-09-25
 */
public class PlainSQLGenerator implements SQLGenerator {
    @Override
    public String getSelectAll(String schema, String tableName) {
        if (StringUtils.isEmpty(schema)) {
            return String.format("SELECT * FROM %s", tableName);
        }
        return String.format("SELECT * FROM %s.%s", schema, tableName);
    }

    @Override
    public String getInsertValues(String schema, String tableName, String values) {
        if (StringUtils.isEmpty(schema)) {
            return String.format("INSERT INTO %s VALUES (%s)", tableName, values);
        }
        return String.format("INSERT INTO %s.%s VALUES (%s)", schema, tableName, values);
    }

    @Override
    public String getRowCount(String schema, String tableName) {
        if (StringUtils.isEmpty(schema)) {
            return String.format("SELECT count(1) FROM %s", tableName);
        }
        return String.format("SELECT count(1) FROM %s.%s", schema, tableName);
    }
}

File: src/main/kotlin/com/koldyr/csv/db/PlainSQLGenerator.kt
package com.koldyr.csv.db

import org.apache.commons.lang3.StringUtils.isEmpty

/**
 * Description of class PlainSQLGenerator
 *
 * @created: 2018-09-25
 */
class PlainSQLGenerator : SQLGenerator {
    override fun getSelectAll(schema: String, tableName: String): String {
        return if (isEmpty(schema)) {
            "SELECT * FROM $tableName"
        } else "SELECT * FROM $schema.$tableName"
    }

    override fun getInsertValues(schema: String, tableName: String, values: String): String {
        return if (isEmpty(schema)) {
            "INSERT INTO $tableName VALUES ($values)"
        } else "INSERT INTO $schema.$tableName VALUES ($values)"
    }

    override fun getRowCount(schema: String, tableName: String): String {
        return if (isEmpty(schema)) {
            "SELECT count(1) FROM $tableName"
        } else "SELECT count(1) FROM $schema.$tableName"
    }
}
